### ISSUE TITLE
build failure: build/release/src/build.c: In function 'fixup_length_block'

### ISSUE THREAD

**Issue by noptrix** [zznop/flyr#21]
Ok, another issue. :)

```
--- snip ---
build/release/src/build.c: In function 'fixup_length_block':
build/release/src/build.c:50:24: error: implicit declaration of function '__bswap16'; did you mean '__bswap_16'? [-Werror=implicit-function-declaration]
                 word = __bswap16(word) + size;
                        ^~~~~~~~~
                        __bswap_16
build/release/src/build.c:64:25: error: implicit declaration of function '__bswap32'; did you mean '__bswap_32'? [-Werror=implicit-function-declaration]
                 dword = __bswap32(dword) + size;
                         ^~~~~~~~~
                         __bswap_32
build/release/src/build.c:77:25: error: implicit declaration of function '__bswap64'; did you mean '__bswap_64'? [-Werror=implicit-function-declaration]
                 qword = __bswap64(qword) + size;
                         ^~~~~~~~~
                         __bswap_64
cc1: all warnings being treated as errors
scons: *** [build/release/src/build.o] Error 1
scons: building terminated because of errors.
--- snip ---
```

**Comment by zznop** [comment]
Try to pull on master and build again. I added `#include <endian.h>` to build.c in hopes that it addresses your issue. `__bswap*` functions are also supposed to be gcc builtins. Depending on your version of gcc, that may not be the case. It builds for me on Debian 9 and on Windows under Cygwin. Also, it builds under docker with the dockerfile included in the repository. Let me know if you're still having issues. I may need to change them to `__builtin_bswap*`

**Comment by noptrix** [comment]
@zznop yes, sorry. due to time pressure here i forgot to add few more details like os, gcc ver, etc.: so've i used:
- archlinux (kernel 4.18.16)
- gcc: 8.2.1
- glibc: 2.28
- scons: 3.0.1
I've managed to add a temporary/dirty fix: https://github.com/BlackArch/blackarch/blob/feadc0bdd58f330c29d3d67c2e15158ef8c4e0f5/packages/flyr/PKGBUILD#L27 so i guess for newer versions you will need '__bswap_*'. in the meanwhile i will use '#include <endian.h>'  in build.c and see if it solved that issue.

**Comment by zznop** [comment]
I’ll build a BlackArch Linux VM this week sometime and look into a fix that keeps support for Debian/Ubuntu. This project is under rapid development so I’d hate for you to have to go through the trouble for each release. I can probably check if `__bswap_*` functions are defined and then typedef the `__bswap*` functions (if including endian.h doesn’t do the trick on Arch). I’m excited to be apart of your distro!

**Comment by noptrix** [comment]
> I’ll build a BlackArch Linux VM this week sometime and look into a fix that keeps support for Debian/Ubuntu. This project is under rapid development so I’d hate for you to have to go through the trouble for each release. I can probably check if `__bswap_*` functions are defined and then typedef the `__bswap*` functions (if including endian.h doesn’t do the trick on Arch). I’m excited to be apart of your distro!

Sounds good and thanks a lot! Unfortunately, ```#include <endian.h``` did not help, jfyi. Yes, typedef should help us. :)

**Comment by aedrax** [comment]
same issue here, but changing `__bswap*` to `__bswap_*` immediately fixes the build issues.

* ubuntu 18.04 (kernel 4.15.0-36)
* gcc: 7.3.0
* glibc: 2.27
* scons: 3.0.1

was about to make a pull request but I saw it may just be an OS specific issue

**Comment by aedrax** [comment]
I added some simple define lines in `conversion.h` where `byteswap.h` is included to handle this issue
```
//added fix to handle different OS byteswap defines
#ifndef __bswap16
#define __bswap16 __bswap_16
#endif
#ifndef __bswap32
#define __bswap32 __bswap_32
#endif
#ifndef __bswap64
#define __bswap64 __bswap_64
#endif
```

**Comment by zznop** [comment]
I believe this was resolved by #22. Please re-open it @noptrix if you have any issues on BlackArch.
